dbms/src/TestUtils/mockExecutor.cpp

```diff
@@ -0,0 +1,239 @@
+    return dag_request_ptr;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::mockTable(const String & db, const String & table, const MockColumnInfos & columns)
+{
+    assert(!columns.empty());
+    TableInfo table_info;
+    table_info.name = db + "." + table;
+    for (const auto & column : columns)
+    {
+        TiDB::ColumnInfo ret;
+        ret.tp = column.second;
+        ret.name = column.first;
+        table_info.columns.push_back(std::move(ret));
+    }
+    String empty_alias;
+    root = compileTableScan(getExecutorIndex(), table_info, empty_alias, false);
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::mockTable(const MockTableName & name, const std::vector<std::pair<String, TiDB::TP>> & columns)
+{
+    return mockTable(name.first, name.second, columns);
+}
+
+DAGRequestBuilder & DAGRequestBuilder::mockTable(const MockTableName & name, const MockColumnInfoList & columns)
+{
+    return mockTable(name.first, name.second, columns);
+}
+
+DAGRequestBuilder & DAGRequestBuilder::filter(ASTPtr filter_expr)
+{
+    assert(root);
+    root = compileSelection(root, getExecutorIndex(), filter_expr);
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::limit(int limit)
+{
+    assert(root);
+    root = compileLimit(root, getExecutorIndex(), buildLiteral(Field(static_cast<UInt64>(limit))));
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::limit(ASTPtr limit_expr)
+{
+    assert(root);
+    root = compileLimit(root, getExecutorIndex(), limit_expr);
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::topN(ASTPtr order_exprs, ASTPtr limit_expr)
+{
+    assert(root);
+    root = compileTopN(root, getExecutorIndex(), order_exprs, limit_expr);
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::topN(const String & col_name, bool desc, int limit)
+{
+    assert(root);
+    root = compileTopN(root, getExecutorIndex(), buildOrderByItemList({{col_name, desc}}), buildLiteral(Field(static_cast<UInt64>(limit))));
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::topN(MockOrderByItems order_by_items, int limit)
+{
+    return topN(order_by_items, buildLiteral(Field(static_cast<UInt64>(limit))));
+}
+
+DAGRequestBuilder & DAGRequestBuilder::topN(MockOrderByItems order_by_items, ASTPtr limit_expr)
+{
+    assert(root);
+    root = compileTopN(root, getExecutorIndex(), buildOrderByItemList(order_by_items), limit_expr);
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::project(const String & col_name)
+{
+    assert(root);
+    root = compileProject(root, getExecutorIndex(), buildColumn(col_name));
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::project(MockAsts exprs)
+{
+    assert(root);
+    auto exp_list = std::make_shared<ASTExpressionList>();
+    for (const auto & expr : exprs)
+    {
+        if (typeid_cast<const ASTFunction *>(expr.get()))
+        {
+            throw TiFlashTestException(fmt::format("Can't include Function in project."));
```

Comment:
Why is there this restriction?

Reply:
I will remove it, currently don't support agg function in projection.